plugins/com.google.cloud.tools.eclipse.projectselector/src/com/google/cloud/tools/eclipse/projectselector/ProjectRepository.java

```diff
@@ -68,7 +68,10 @@ public ProjectRepository(IGoogleApiFactory apiFactory) {
           listRequest = listRequest.setPageToken(token); 
         }
         ListProjectsResponse response = listRequest.execute();
-        projectList.addAll(response.getProjects());
+        List<Project> responseProjects = response.getProjects();
```

Comment:
This PR seems vastly more complex than I expected. I thought this and a test would be all that was needed.